src/components/MapView/AlertForm/index.tsx

```diff
@@ -0,0 +1,330 @@
+
+// This should probably be determined on a case-by-case basis,
+// depending on if the downstream API has the capability.
+// For now it can be permanently enabled.
+const ALERT_FORM_ENABLED = true;
+
+enum Statistic {
+  Maximum,
+  Minimum,
+  Median,
+  Mean,
+}
+
+enum Comparator {
+  '>',
+  '>=',
+  '<',
+  '<=',
+}
+
+const boundaryLayer = getBoundaryLayerSingleton();
+
+function AlertForm({ classes }: AlertFormProps) {
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const [isAlertFormFormOpen, setIsAlertFormFormOpen] = useState(false);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState<LayerKey>();
+  const [regionsList, setRegionsList] = useState<string[]>(['allRegions']);
+  const [emailValid, setEmailValid] = useState<boolean>(false);
+  const [email, setEmail] = useState('');
+  const [selectedStat, setSelectedStat] = useState<Statistic>();
+  const [selectedComparator, setSelectedComparator] = useState('');
+  const [selectedThreshold, setSelectedThreshold] = useState<number>(0);
+
+  const regionNamesToNsoCodes: { [k: string]: string } = useMemo(() => {
+    if (!boundaryLayerData) {
+      // Not loaded yet. Will proceed when it is.
+      return {};
+    }
+
+    return Object.fromEntries(
+      boundaryLayerData.data.features
+        .filter(feature => feature.properties != null)
+        .map(feature => [
+          `${feature.properties?.ADM1_EN} / ${feature.properties?.ADM2_EN}`,
+          feature.properties?.NSO_CODE,
+        ]),
+    );
+  }, [boundaryLayerData]);
+
+  const sortedRegionNames: string[] = useMemo(() => {
+    // Fine to mutate this array since it's a new array of key names
+
+    // eslint-disable-next-line fp/no-mutating-methods
+    return Object.keys(regionNamesToNsoCodes).sort();
+  }, [regionNamesToNsoCodes]);
+
+  const onChangeEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newEmail = event.target.value;
+    setEmailValid(!!newEmail.match(EMAIL_REGEX));
+    setEmail(newEmail);
+  };
+
+  const runAlertForm = async () => {
+    const regionCodes = regionsList
+      .filter(r => r !== 'allRegions')
+      .map(r => regionNamesToNsoCodes[r]);
+
+    const apiData = {
+      hazardLayerId,
+      statistic: selectedStat,
+      comparator: selectedComparator,
+      threshold: selectedThreshold,
+      regionNsoCodes: regionCodes,
+      email,
+      zones_url: '', // part of the ApiData object; refactor needed
+      geotiff_url: '', // ^
+    };
+
+    console.log(apiData);
+
+    // TODO: Make API call
+    // await fetchApiData('http://localhost:80/alarms', apiData);
+  };
+
+  if (!ALERT_FORM_ENABLED) {
+    return null;
+  }
+
+  return (
+    <div className={classes.alertForm}>
+      <Button
```

Comment:
This dropdown code was copied from `Analyser` right? Maybe we can make a new component which allows anyone to make dropdowns like the Analysis menu?

Reply:
Perhaps, although in my conversations with @ericboucher it sounds like maybe the way to create alerts would ideally not be a dropdown anyway - it might be it's own view entirely. So I'm not sure how useful a refactor like this would be at the moment.